datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -1869,6 +1905,51 @@ mod tests {
             Operator::Multiply,
             DataType::Float64
         );
+
+        // Test integer to interval coercion for temporal arithmetic
+        test_coercion_assymetric_binary_rule!(
+            DataType::Int32,
+            DataType::Timestamp(TimeUnit::Nanosecond, None),
+            Operator::Plus,
+            DataType::Interval(IntervalUnit::DayTime),
+            DataType::Timestamp(TimeUnit::Nanosecond, None)
+        );
+        test_coercion_assymetric_binary_rule!(
+            DataType::Timestamp(TimeUnit::Nanosecond, None),
+            DataType::Int64,
+            Operator::Plus,
+            DataType::Timestamp(TimeUnit::Nanosecond, None),
+            DataType::Interval(IntervalUnit::DayTime)
+        );
+        test_coercion_assymetric_binary_rule!(
+            DataType::Int32,
+            DataType::Date32,
```

Comment:
do we also have Date32 + Int32 test (in that order)?

Reply:
Probably not. Could be done